core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -121,13 +136,30 @@ object FastPassService extends LazyLogging {
                              googleProject,
                              googleIamDAO,
                              googleStorageDAO
-        )
+        ).transform {
+          case Failure(e) =>
+            logger.error(
+              s"Encountered an error while removing FastPass grants for ${accountEmail.value} in ${googleProjectId.value}",
+              e
+            )
+            openTelemetry.incrementCounter("fastpass-removal-failure").unsafeRunSync()
+            Success(Left((e, grants)))
+          case Success(_) => Success(Right(grants))
+        }
     }
     for {
       // this `flatMap` is necessary to run the IAM Updates in sequence, as to not sent conflicting policy updates to GCP
-      _ <- DBIO.from(iamUpdates.foldLeft(Future.successful())((a, b) => a.flatMap(_ => b())))
-      _ <- removeGrantsFromDb(fastPassGrants.map(_.id), dataAccess, optCtx)
-    } yield ()
+      iamRemovals <-
+        DBIO.from(
+          iamUpdates
+            .foldLeft(Future.successful[Seq[RemovalResult]](Seq.empty[RemovalResult]))((a, b) =>
+              a.flatMap(results => b().map(_ +: results))
+            )
+        )
+      successfulIamRemovals = iamRemovals.collect { case Right(seq) => seq }.flatten
+      failedIamRemovals = iamRemovals.collect { case Left(e) => e }
+      _ <- removeGrantsFromDb(successfulIamRemovals.map(_.id), dataAccess, optCtx)
+    } yield failedIamRemovals
```

Comment:
Yeah, this is really messy, but in a way scala makes hard to untangle. Dealing with concurrency and collecting errors is always a sore spot, so I'm using the standard Scala `Either` library to collect results and errors. 

The way we have to `flatmap` over the updates is because we need to force the Futures to execute sequentially. This, combined with the error handling, makes it really hard to refactor for readability. 

I did my best to pull the type out so that it wasn't muddying the waters, but I think this is just the result of 2 "hard" problems in software and Scala's hard-to-read-but-idiomatic way of solving it.